From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 9 Oct 2019 11:36:56 -0700
Subject: [PATCH] md_internal.h: Added <string> C++ header API

This helps in compiling with libc++/Clang
Fixes
usr/include/c++/v1/type_traits:1561:38: error: implicit instantiation
of undefined template 'std::__1::hash<std::__1::basic_string<char,
std::__1::char_traits<char>, std::__1::allocator<char> > >'
: public integral_constant<bool, __is_empty(_Tp)> {};
^

Change-Id: Ib48031bc25dd95e71748283161f2ef9d5db6e2c2
Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Andrzej Datczuk <andrzej.datczuk@intel.com>
Upstream-Status: Backport
Reference: https://github.com/intel/metrics-discovery/commit/c3d30164af
---
 instrumentation/metrics_discovery/common/md_internal.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/instrumentation/metrics_discovery/common/md_internal.cpp b/instrumentation/metrics_discovery/common/md_internal.cpp
index f9571ac..784136e 100644
--- a/instrumentation/metrics_discovery/common/md_internal.cpp
+++ b/instrumentation/metrics_discovery/common/md_internal.cpp
@@ -29,6 +29,7 @@
 
 #include <string.h>
 #include <stdlib.h>
+#include <string>
 #include <new>
 #include <unordered_map>
 
